src/components/common/UserProfileEdit/UserAdvanceSettings.tsx

```diff
@@ -0,0 +1,243 @@
+    ? metatransactionsSetting
+    : false;
+
+  return (
+    <Form<FormValues>
+      initialValues={{
+        [SlotKey.Metatransactions]: metatransasctionsAvailable,
+        [SlotKey.DecentralizedMode]: decentralizedModeEnabled,
+        [SlotKey.CustomRPC]: customRpc,
+      }}
+      validationSchema={validationSchema}
+      onSubmit={onSubmit}
+    >
+      {({ isValid, values, submitForm }) => (
+        <div className={styles.main}>
+          <Heading
+            appearance={{ theme: 'dark', size: 'medium' }}
+            text={MSG.heading}
+            textValues={{
+              // @ts-ignore
+              learnMoreLink: (
+                <ExternalLink
+                  text={{ id: 'text.learnMore' }}
+                  href=""
+                  className={stylesAdvance.link}
+                />
+              ),
+            }}
+          />
+          <div className={stylesAdvance.toggleContainer}>
+            <Toggle
+              label={MSG.labelMetaTx}
+              labelValues={{
+                isOn: values[SlotKey.Metatransactions],
+              }}
+              name={SlotKey.Metatransactions}
+              disabled={!metatransasctionsToggleAvailable}
+            />
+            <QuestionMarkTooltip
+              tooltipText={MSG.metaTooltip}
+              /* @ts-ignore */
+              tooltipTextValues={{ br: <br /> }}
+              className={stylesAdvance.tooltipContainer}
+              tooltipClassName={stylesAdvance.tooltipContent}
+              tooltipPopperOptions={{
+                placement: 'right',
+              }}
+            />
+          </div>
+          <p className={stylesAdvance.descriptions}>
+            <FormattedMessage {...MSG.metaDescription} />
+          </p>
+          {!metatransasctionsToggleAvailable && (
+            <div className={stylesAdvance.metaDesc}>
+              <FormattedMessage {...MSG.metaDescGlobalOff} />
+            </div>
+          )}
+          <hr />
+          <div className={stylesAdvance.sectionTitle}>
+            <FormattedMessage
+              {...MSG.customEndpoints}
+              values={{ isOn: values[SlotKey.DecentralizedMode] }}
+            />
+          </div>
+          <p className={stylesAdvance.descriptions}>
+            <FormattedMessage {...MSG.metaDescription} />
+          </p>
+          <div className={stylesAdvance.toggleContainer}>
+            <Toggle
+              label={MSG.labelGnosisRPC}
+              labelValues={{ isOn: decentralizedModeEnabled }}
+              name={SlotKey.DecentralizedMode}
+            />
+            <QuestionMarkTooltip
+              tooltipText={MSG.gnosisRPCTooltip}
+              className={stylesAdvance.tooltipContainer}
+              tooltipClassName={stylesAdvance.tooltipContent}
+              tooltipPopperOptions={{
+                placement: 'right',
+              }}
+            />
+          </div>
+          <Input
+            name={SlotKey.CustomRPC}
```

Comment:
Since this is an input specific to a url, we could specify formatting options to prevent the user from inputting whitespace. I copied a url from the browser and it came with a bunch of whitespace at the end, which resulted in the "invalid url" error showing, which was confusing given that I'd entered a valid url (but of course, was because of the trailing whitespace).